### ISSUE TITLE
Creating group without permissions causes segfault

### ISSUE THREAD

**Issue by rexxars** [sanity-io/gradient#335]
If you insert a group (`system.group` document in `_.groups`) and forget to specify `permissions` for a grant, Gradient segfaults:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5da03c]

goroutine 71 [running]:
panic(0xc58d20, 0xc420012060)
 /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/sanity-io/gradient/pkg/security.(*Permissions).Copy(0x0, 0xc42031f860)
 /go/src/github.com/sanity-io/gradient/pkg/security/permissions.go:132 +0x6c
github.com/sanity-io/gradient/pkg/security.(*grantCacheNode).addMember(0xc4206e2f80, 0xc42075e400, 0x18, 0xc420245fb0, 0x1, 0x1, 0x1, 0x0)
 /go/src/github.com/sanity-io/gradient/pkg/security/grant_cache.go:150 +0x4fe
github.com/sanity-io/gradient/pkg/security.(*GrantCache).AddMember(0xc42024a228, 0xc42075e400, 0x18, 0xc420245fb0, 0x1, 0x1, 0x0)
 /go/src/github.com/sanity-io/gradient/pkg/security/grant_cache.go:41 +0xe4
github.com/sanity-io/gradient/pkg/system.(*GroupCache).AddGroup(0xc420715fc0, 0xc4206e2f00)
 /go/src/github.com/sanity-io/gradient/pkg/system/group_cache.go:103 +0x104
github.com/sanity-io/gradient/pkg/system.(*namespace).AddGroup(0xc4201ea4b0, 0xc4206e2f00)
 /go/src/github.com/sanity-io/gradient/pkg/system/namespace.go:111 +0x39
github.com/sanity-io/gradient/pkg/system.(*Manager).addedGroup(0xc4201bb900, 0xc4206d3c20, 0x1e, 0xc4206d3c40, 0x16, 0xc42031f5c0, 0xc4206d3e20, 0x16, 0x1, 0xc4206d3c49, ...)
 /go/src/github.com/sanity-io/gradient/pkg/system/manager.go:550 +0x131
github.com/sanity-io/gradient/pkg/system.(*Manager).addedModel(0xc4201bb900, 0xc4206d3c20, 0x1e, 0xc4206d3c40, 0x16, 0xc42031f5c0, 0xc4206d3e20, 0x16, 0x1, 0x0, ...)
 /go/src/github.com/sanity-io/gradient/pkg/system/manager.go:493 +0x6bb
github.com/sanity-io/gradient/pkg/system.(*Manager).runEventCh(0xc4201bb900, 0xc4201f5aa0)
 /go/src/github.com/sanity-io/gradient/pkg/system/manager.go:446 +0x2ec
created by github.com/sanity-io/gradient/pkg/system.(*Manager).Start
 /go/src/github.com/sanity-io/gradient/pkg/system/manager.go:156 +0x65d
```

It looks like the document is still stored in database, which causes Gradient to crash on startup when it picks up the document again.

**Comment by erikgrinaker** [comment]
This seems to have been halfway fixed. I was able to create a group without specifying `permissions` under `grants`, but when trying to replace the group I get a panic. Example:

```bash
$ curl -sSfv -XPOST localhost:56200/mutate/grouptest -d '{
 	"mutations": [
		{"createOrReplace": {
			"_id": "_.groups.everyone",
			"_type": "system.group",
			"members": ["everyone"],
			"grants": [
				{
					"path": "*"
				}
			]
		}}
	]
}'
```

This works on the first run, but the second run yields a 500:

```
12:30:58.914 CRITICAL server.httpapi.http.[6thP5mXd58sml53MTXhZzy] Error (runtime.errorString): Panic in HTTP handler while processing POST /mutate/grouptest: runtime error: invalid memory address or nil pointer dereference
github.com/sanity-io/gradient/pkg/security.(*Permissions).Has(...)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/security/permissions.go:85
github.com/sanity-io/gradient/pkg/security.(*GrantCache).generateGlobalFilterExpressionForIdentityForPermission(0xc420387d10, 0x1ab1233, 0xb, 0x1aab4a4, 0x4, 0x0, 0x10, 0x1929820, 0xc4201bb8c0)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/security/grant_cache.go:197 +0xaf
github.com/sanity-io/gradient/pkg/security.(*GrantCache).generateGlobalFilterSetForIdentity(0xc420387d10, 0x1ab1233, 0xb, 0xb, 0x220a320, 0x0)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/security/grant_cache.go:182 +0x188
github.com/sanity-io/gradient/pkg/security.(*GrantCache).GlobalFilterFor(0xc420387d10, 0x1ab1233, 0xb, 0x1aacff5, 0x6, 0xc420178369, 0x8, 0x1045101, 0x30)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/security/grant_cache.go:96 +0xc0
github.com/sanity-io/gradient/pkg/system.(*GroupCache).GlobalFilterFor(0xc4203b9e40, 0x21d63a0, 0x1aacff5, 0x6, 0xc420062701, 0xc4203c1770, 0xc4203aa210, 0x10d91ab)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/system/group_cache.go:40 +0x27c
github.com/sanity-io/gradient/pkg/system.(*GroupCache).HasPermissionOnDocument(0xc4203b9e40, 0xc420441297, 0x9, 0xc420178360, 0x11, 0xc4203c1590, 0x0, 0x0, 0x0, 0x21d63a0, ...)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/system/group_cache.go:80 +0xc4
github.com/sanity-io/gradient/pkg/system.(*session).HasPermissionOnDocument(0xc4203e4360, 0xc420441297, 0x9, 0xc420178360, 0x11, 0xc4203c1590, 0x0, 0x0, 0x0, 0x1aacff5, ...)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/system/session.go:271 +0x216
github.com/sanity-io/gradient/pkg/system.(*permissionCheckingMiddleware).OnMutation(0xc420174d40, 0xc420346690, 0x0, 0x0)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/system/permission_checking_middleware.go:38 +0x28d
github.com/sanity-io/gradient/pkg/documents/middlewares.chain.OnMutation(0xc4203c15f0, 0x3, 0x3, 0xc420346690, 0x111db01, 0xc420346690)
	/Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/documents/middlewares/package.go:21 +0x6b
github.com/sanity-io/gradient/pkg/documents/middlewares.(*chain).OnMutation(0xc4203e4960, 0xc420346690, 0xc420178360, 0x11)
	<autogenerated>:1 +0x58
```

Investigating.

**Comment by erikgrinaker** [comment]
I've committed a fix for the `nil` pointer panic in 76bb309b26c3349f70f7542ac46a7cec78c2b06a.

It probably shouldn't be possible to create groups without specifying an explicit permissions list, I've opened PR #499 for this.
